pandas/_libs/algos_groupby_helper.pxi.in

```diff
@@ -55,37 +56,65 @@ def group_add_{{name}}(ndarray[{{dest_type2}}, ndim=2] out,
 
     with nogil:
 
-        if K > 1:
+        if skipna == False:
+            if K > 1:
 
-            for i in range(N):
-                lab = labels[i]
-                if lab < 0:
-                    continue
-
-                counts[lab] += 1
-                for j in range(K):
-                    val = values[i, j]
+                for i in range(N):
```

Comment:
so I think you are better off making only a single block (like it was), 

then checking the flag like this (around like 69-70)

(rename skipna -> checknull as that is more consistent with what we use)
```
if checknull and val == val:
```

separately, you can pass this flag  (see cummin) as well (so you are adding 2 args)
also have a look here
https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/groupby_helper.pxi.in#L597

to deal with datetimelikes (basically this is how we check for nulls there, the ``val==val`` trick only works for floats).